### ISSUE TITLE
Possible for Pokemon to evolve into variants of themselves?

### ISSUE THREAD

**Issue by Durandbolg** [Rangi42/polishedcrystal#756]
tldr:
I'd love to have the ability for Pokemon to evolve into variants of themselves (for my own hack). Is that something that's feasible?

Full:
Hello, there! I'm sorry if this isn't the right place to ask this, but I was looking into building my own hack off of the existing polished crystal data, but had a specific question about something that I was looking into trying. 

Is it possible to have a Pokemon evolve into another form of itself? For example, to test out the idea, I took a random variant (alolan sandshrew) and tried to make it evolve into the plain version of itself (using an evolution item. Any would do for testing purposes), but it just stayed in the alolan form. (The evolution event played as normal though, so I know it at least went through.)

I know very little about coding, but if I had to guess, maybe it just has the alolan form called up, but there's no instructions that tells it specificially to change to the plain form, so it just keeps the currently called variable? The only place I see so far to alter evolutions is in evos_attacks.asm, and cygwin gives me an error if I try to put _plain next to sandshrew. (See below)

SandshrewAlolanEvosAttacks:
	db EVOLVE_ITEM, FIRE_STONE, SANDSHREW

**Comment by Rangi42** [comment]
Try explicitly specifying `db EVOLVE_ITEM, FIRE_STONE, SANDSHREW, PLAIN_FORM`. If that doesn't work, then something about the current code just doesn't support it, and you'd have to edit the code to make it work. But I think it will work.

**Comment by FredrIQ** [comment]
> Try explicitly specifying `db EVOLVE_ITEM, FIRE_STONE, SANDSHREW, PLAIN_FORM`. If that doesn't work, then something about the current code just doesn't support it, and you'd have to edit the code to make it work. But I think it will work.

This does not work in current master (and how could it work; how would the game know the difference between explicit form and lack of it?).

The future 9bit update should have proper support for this. In current master, you would need to add a special case in the evolution code for it.

**Comment by Durandbolg** [comment]
Thanks for the replies! The only reason I ask is because if it WERE possible, that would mean that one could squeeze in the baby pokemon as special "infant forms" that bear the name of their second stage form, but evolve into it as normal (So jigglypuff_infant would evolve into jigglypuff_plain, for instance). Not an elegant solution, as the possible byte extension is more permanent, but I thought I'd give it a try. Ah, well.
